astropy/coordinates/angles.py

```diff
@@ -660,7 +674,14 @@ def wrap_angle(self):
 
     @wrap_angle.setter
     def wrap_angle(self, value):
-        self._wrap_angle = Angle(value)
+        if not isinstance(value, Angle):
```

Comment:
I'm inclined to agree.  I'd rather just see `Angle(value)` for `isinstance(value, Angle)` sped up.